keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -514,11 +514,21 @@ def compute_metrics(self, x, y, y_pred, sample_weight):
                 metrics[metric.name] = result
         return metrics
 
+    @classmethod
+    def from_config(cls, config):
+        config["backbone"] = keras.utils.deserialize_keras_object(
+            config["backbone"]
+        )
+        return super().from_config(config)
+
     def get_config(self):
         return {
             "num_classes": self.num_classes,
             "bounding_box_format": self.bounding_box_format,
-            "backbone": self.backbone,
+            "backbone": keras.utils.serialize_keras_object(self.backbone),
```

Comment:
I didn't realize that serialization also needs to be explicit, I thought it was just deserialization?

But either way this LGTM, as this seems like the most defensive approach.

Reply:
Only deserialization should be explicitly necessary here. But this seems like a good defensive approach either way! Approved.

Reply:
This actually breaks old TF versions, we should bump to 2.12

Reply:
That would affect the experience of Colab users. We need to make sure Colab is updated to TF 2.12 when keras-cv 0.5 is released.